config/config.go

```diff
@@ -310,7 +310,7 @@ type ErrConfigValidationFailed struct {
 }
 
 func (e *ErrConfigValidationFailed) Error() string {
-	return fmt.Sprintf("config file %s contained unknown configuration options: %s", e.confFile, strings.Join(e.UndecodedItems, ", "))
+	return fmt.Sprintf("config file %s contained deprecated configuration options: %s", e.confFile, strings.Join(e.UndecodedItems, ", "))
```

Comment:
How does it kow the difference between an invalid/unknown and a deprecated configuration variable? I would argue at this point the option has not been deprecated -- it's actually been removed.

If you wanted to cover both cases it might be better to say:

"config file %s contained invalid configuration options: %s. If this does not appear to be a typo, check the TiDB manual to make sure this option has not been deprecated and removed from your TiDB version."

Reply:
done, thx

PTAL

Reply:
LGTM